build_tools/cmake/clean_build.sh

```diff
@@ -0,0 +1,26 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Build the IREE project with CMake. Designed for CI, but can be run manually.
+# This is equivalent to the build_tools/cmake/rebuild.sh script except it
+# first deletes the build directory to deal with any faulty caching.
```

Comment:
Wouldn't we want to use caching on CI if possible? If we switch from on-demand VM instances to persistent instances that keep local caches, we could see some big speedups to Bazel and CMake...

Are there specific instances of "faulty caching" that we've seen? I'd like to understand any such cases better.

How do other CMake projects handle caching on CI?

Reply:
We haven't up to this point, so I think that should be a separate decision. For now I just wanted to stop copy-pasting cmake commands from the script :-D

I seem to recall faulty caching at some point previously when working locally, but maybe that was when build options changed and rebuilding from modified source with the same options would be fine.

@marbre, what's your experience with cmake caching?

Reply:
Well the cmake projects where I had a CI available, I always triggered clean builds. Caching does works most times from my experiences, but I already had several cases locally where something wasn't rebuild as I expected. As a result, the clean build was broken which was only discovered because the build in the CI was broken... Furthermore, I am seeing `Illegal Glob` errer messages or something like this (cannot remember the exact message) locally from time to time, which might be a problem with the way we add the lit tests. So far the only thing that worked out for me to solve this was to perform a clean build...

Reply:
Yeah this is confirming my reluctance to try and use cmake caching